examples/custom-node/src/primitives/extended_op_tx_envelope.rs

```diff
@@ -0,0 +1,327 @@
+use alloy_consensus::Transaction;
+use alloy_eips::{
+    eip2718::{Eip2718Error, Eip2718Result},
+    eip2930::AccessList,
+    eip7702::SignedAuthorization,
+    Decodable2718, Encodable2718, Typed2718,
+};
+use alloy_primitives::{ChainId, PrimitiveSignature, TxHash};
+use alloy_rlp::{BufMut, Decodable, Encodable, Result as RlpResult};
+use op_alloy_consensus::{OpTxEnvelope, OpTxType};
+use reth_codecs::Compact;
+use reth_primitives_traits::{transaction::signed::RecoveryError, InMemorySize, SignedTransaction};
+use revm_primitives::{Address, Bytes, TxKind, B256, U256};
+use std::convert::TryInto;
+
+macro_rules! delegate {
+    ($self:expr => $tx:ident.$method:ident($($arg:expr),*)) => {
+        match $self {
+            Self::BuiltIn($tx) => $tx.$method($($arg),*),
+            Self::Other($tx) => $tx.$method($($arg),*),
+        }
+    };
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, Hash, Eq, PartialEq)]
+pub enum ExtendedOpTxEnvelope<T> {
+    BuiltIn(Box<OpTxEnvelope>),
+    Other(T),
+}
+
+// TODO: add SerdeBincodeCompat impl once https://github.com/paradigmxyz/reth/issues/15377 is done
+// #[derive(Debug)]
+// pub enum ExtendedOpTxEnvelopeRepr<'a, T: SerdeBincodeCompat> {
+//     BuiltIn(<OpTxEnvelope as SerdeBincodeCompat>::BincodeRepr<'a>),
+//     Other(T::BincodeRepr<'a>),
+// }
+//
```

Comment:
pls move this down to the same place as its impl body

Reply:
good point, done here https://github.com/paradigmxyz/reth/pull/15334/commits/f66b8836c81581131d88a34f31c6d9e123eccb0d